### ISSUE TITLE
model doesn't satisfy

### ISSUE THREAD

**Issue by akgperson** [CVC4/CVC4#2153]
Error message:
```
(error "Internal error detected
void CVC4::TheoryEngine::checkTheoryAssertionsWithModel(bool)
../../../../src/theory/theory_engine.cpp:2275:

  
THEORY_ARITH has an asserted fact that the model doesn't satisfy.
The fact: (= kf_10 (+ (* (- 1.0) (* r0 r15)) (* (/ 2000 681247) (* r0 r0 r15)) (* (/ (- 1000) 681247) (* r0 r5 r15)) (* (/ (- 1000) 681247) (* r0 r15 r15)) (* (/ 1000 681247) (* r0 r15 r16))))
Model value: false
")

```
Input File:
```
(set-logic ALL)
(set-option :incremental true)
(declare-const v0 Bool)
(declare-const v1 Bool)
(declare-const v2 Bool)
(declare-const v3 Bool)
(declare-const v4 Bool)
(declare-const v5 Bool)
(declare-const i1 Int)
(declare-const i5 Int)
(declare-const i6 Int)
(declare-const i7 Int)
(declare-const r0 Real)
(declare-const r2 Real)
(declare-const r3 Real)
(declare-const r4 Real)
(declare-const r5 Real)
(declare-const r7 Real)
(declare-const r10 Real)
(declare-const r11 Real)
(declare-const r13 Real)
(declare-const r14 Real)
(declare-sort S0 0)
(declare-const v6 Bool)
(declare-const v7 Bool)
(declare-const i10 Int)
(declare-const r15 Real)
(declare-const v8 Bool)
(declare-const i11 Int)
(declare-const i12 Int)
(declare-const v9 Bool)
(declare-const v10 Bool)
(declare-const r16 Real)
(declare-const v11 Bool)
(declare-const r17 Real)
(declare-const v12 Bool)
(declare-const v13 Bool)
(declare-const r18 Real)
(declare-const v14 Bool)
(declare-const i13 Int)
(declare-const r19 Real)
(declare-const r20 Real)
(declare-const v15 Bool)
(declare-const v16 Bool)
(declare-const v17 Bool)
(declare-const r21 Real)
(declare-sort S1 0)
(declare-const v18 Bool)
(declare-const r22 Real)
(declare-const v19 Bool)
(declare-const r23 Real)
(declare-const v20 Bool)
(declare-const S1-0 S1)
(declare-const v21 Bool)
(declare-const r24 Real)
(declare-const v22 Bool)
(declare-const v23 Bool)
(declare-sort S2 0)
(declare-const i14 Int)
(declare-sort S3 0)
(declare-const v24 Bool)
(declare-const v25 Bool)
(declare-const v26 Bool)
(declare-const v27 Bool)
(declare-const r25 Real)
(declare-const S3-0 S3)
(declare-const v28 Bool)
(declare-const S0-0 S0)
(declare-const v29 Bool)
(declare-const v30 Bool)
(declare-const v31 Bool)
(declare-const v32 Bool)
(declare-const v33 Bool)
(declare-const r26 Real)
(declare-const v34 Bool)
(declare-const r27 Real)
(declare-const v35 Bool)
(declare-const v36 Bool)
(declare-const v37 Bool)
(declare-const i15 Int)
(declare-const S0-1 S0)
(declare-const r28 Real)
(declare-const v38 Bool)
(declare-const r29 Real)
(declare-const v39 Bool)
(declare-const S3-1 S3)
(declare-const v40 Bool)
(declare-const v41 Bool)
(declare-const v42 Bool)
(declare-const v43 Bool)
(declare-const S0-2 S0)
(declare-const v44 Bool)
(declare-const v45 Bool)
(declare-const v46 Bool)
(declare-const r30 Real)
(declare-const v47 Bool)
(declare-const i16 Int)
(declare-const v48 Bool)
(declare-const v49 Bool)
(declare-const v50 Bool)
(declare-const i17 Int)
(declare-const r31 Real)
(declare-const r32 Real)
(declare-const v51 Bool)
(declare-const v52 Bool)
(declare-const i18 Int)
(declare-const S1-1 S1)
(declare-const v53 Bool)
(declare-const v54 Bool)
(declare-const v55 Bool)
(declare-const v56 Bool)
(declare-const v57 Bool)
(declare-const v58 Bool)
(declare-const v59 Bool)
(declare-const r33 Real)
(declare-const v60 Bool)
(declare-const v61 Bool)
(declare-const i19 Int)
(declare-const r34 Real)
(declare-const r35 Real)
(declare-const v62 Bool)
(declare-const r36 Real)
(declare-const S2-0 S2)
(declare-const S2-1 S2)
(declare-const i20 Int)
(declare-const r37 Real)
(declare-const i21 Int)
(declare-const r38 Real)
(declare-const r39 Real)
(declare-const v63 Bool)
(declare-const v64 Bool)
(declare-const i22 Int)
(declare-const v65 Bool)
(declare-const S3-2 S3)
(declare-const S2-2 S2)
(declare-const v66 Bool)
(declare-const v67 Bool)
(declare-const r40 Real)
(declare-const v68 Bool)
(declare-const S0-3 S0)
(declare-const v69 Bool)
(declare-const v70 Bool)
(declare-const i23 Int)
(declare-const v71 Bool)
(declare-const v72 Bool)
(declare-const v73 Bool)
(declare-const r41 Real)
(declare-const r42 Real)
(declare-const S3-3 S3)
(declare-const r43 Real)
(declare-const v74 Bool)
(declare-const r44 Real)
(declare-const r45 Real)
(declare-const v75 Bool)
(declare-const r46 Real)
(declare-const S1-2 S1)
(declare-const v76 Bool)
(declare-const i24 Int)
(declare-const v77 Bool)
(declare-const r47 Real)
(declare-const v78 Bool)
(declare-const S1-3 S1)
(declare-const v79 Bool)
(declare-const v80 Bool)
(declare-const v81 Bool)
(declare-const v82 Bool)
(declare-const S3-4 S3)
(declare-const i25 Int)
(declare-const r48 Real)
(declare-const v83 Bool)
(declare-const r49 Real)
(declare-const v84 Bool)
(declare-const r50 Real)
(declare-const v85 Bool)
(declare-const i26 Int)
(declare-const i27 Int)
(declare-const v86 Bool)
(declare-const r51 Real)
(declare-const v87 Bool)
(declare-const i28 Int)
(declare-const r52 Real)
(declare-const v88 Bool)
(declare-const r53 Real)
(declare-const S2-3 S2)
(declare-const v89 Bool)
(declare-const i29 Int)
(declare-const v90 Bool)
(declare-const S2-4 S2)
(declare-const v91 Bool)
(declare-const v92 Bool)
(declare-const S0-4 S0)
(declare-const v93 Bool)
(declare-const v94 Bool)
(declare-const v95 Bool)
(declare-const r54 Real)
(declare-const v96 Bool)
(declare-const v97 Bool)
(declare-const i30 Int)
(declare-const r55 Real)
(declare-const v98 Bool)
(declare-const r56 Real)
(declare-const v99 Bool)
(declare-const i31 Int)
(declare-const r57 Real)
(declare-const v100 Bool)
(declare-const r58 Real)
(declare-const S0-5 S0)
(declare-const v101 Bool)
(declare-const S1-4 S1)
(declare-const v102 Bool)
(declare-const v103 Bool)
(declare-const r59 Real)
(declare-const v104 Bool)
(declare-const v105 Bool)
(declare-const r60 Real)
(declare-const v106 Bool)
(declare-const i32 Int)
(declare-const r61 Real)
(declare-const v107 Bool)
(declare-const v108 Bool)
(declare-const v109 Bool)
(declare-const r62 Real)
(declare-const v110 Bool)
(declare-const v111 Bool)
(declare-const v112 Bool)
(declare-const S0-6 S0)
(assert (or (is_int r22) v27 (= S1-0 S1-0 S1-0)))
(assert (or (<= (- r10 (- (/ (- 25.247) r16)) r3) 74 (* (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (/ 25.247 (- r2)) r15 (- r0) (to_real (abs 92)))) v25 (= S1-0 S1-0 S1-0)))
(assert (or (is_int 3110) (= S1-0 S1-0 S1-0) (and (< 182 (- 58)) v3 (>= (mod 83 i7) 795) (= (/ r5 r4) (- r11)) (>= r19 (+ r14 (- 3110 r3)) 0.11 r4) (distinct 4695330 r3 (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (- r2) (* 176 (- r10 (- (/ (- 25.247) r16)) r3))) v48 (distinct (div (to_int 0.1618) 17) (abs 58)) (<= (to_real (- 592 (abs 61) 182)) (to_real (mod (to_int (+ 85 (- r10 85 (+ 25.247 656 r5 r15)) (+ (- r2) (- 25.247)))) (+ (div (+ 61 i6) (to_int 0.1618)) 708 (mod 61 i5) i12 (to_int 4695330)))) r10 r29))))
(assert (or (> (+ r14 (- 3110 r3)) (/ r28 (- 0.1618))) (is_int r22) v25))
(assert (or (distinct (- (+ 25.247 656 r5 r15)) (- r0) (/ (- 25.247) r16) 0.11 (- (- r2) (/ (- r0) r17) (- r11))) (is_int r22) (is_int 3110)))
(assert (or (<= (- r10 (- (/ (- 25.247) r16)) r3) 74 (* (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (/ 25.247 (- r2)) r15 (- r0) (to_real (abs 92)))) (= S1-0 S1-0 S1-0) (= S1-0 S1-0 S1-0)))
(assert (or (= S1-0 S1-0 S1-0) (is_int r22) (is_int 3110)))
(assert (or (= S1-0 S1-0 S1-0) (distinct (- (+ 25.247 656 r5 r15)) (- r0) (/ (- 25.247) r16) 0.11 (- (- r2) (/ (- r0) r17) (- r11))) (distinct (- (+ 25.247 656 r5 r15)) (- r0) (/ (- 25.247) r16) 0.11 (- (- r2) (/ (- r0) r17) (- r11)))))
(assert (or (= S1-0 S1-0 S1-0) v25 v27))
(assert (or (is_int r22) (and (< 182 (- 58)) v3 (>= (mod 83 i7) 795) (= (/ r5 r4) (- r11)) (>= r19 (+ r14 (- 3110 r3)) 0.11 r4) (distinct 4695330 r3 (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (- r2) (* 176 (- r10 (- (/ (- 25.247) r16)) r3))) v48 (distinct (div (to_int 0.1618) 17) (abs 58)) (<= (to_real (- 592 (abs 61) 182)) (to_real (mod (to_int (+ 85 (- r10 85 (+ 25.247 656 r5 r15)) (+ (- r2) (- 25.247)))) (+ (div (+ 61 i6) (to_int 0.1618)) 708 (mod 61 i5) i12 (to_int 4695330)))) r10 r29)) (> (+ r14 (- 3110 r3)) (/ r28 (- 0.1618)))))
(assert (or (is_int 3110) (distinct (- (+ 25.247 656 r5 r15)) (- r0) (/ (- 25.247) r16) 0.11 (- (- r2) (/ (- r0) r17) (- r11))) (= S1-0 S1-0 S1-0)))
(assert (or (<= (- r10 (- (/ (- 25.247) r16)) r3) 74 (* (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (/ 25.247 (- r2)) r15 (- r0) (to_real (abs 92)))) (distinct (- (+ 25.247 656 r5 r15)) (- r0) (/ (- 25.247) r16) 0.11 (- (- r2) (/ (- r0) r17) (- r11))) (distinct (- (+ 25.247 656 r5 r15)) (- r0) (/ (- 25.247) r16) 0.11 (- (- r2) (/ (- r0) r17) (- r11)))))
(assert (or v25 (<= (- r10 (- (/ (- 25.247) r16)) r3) 74 (* (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (/ 25.247 (- r2)) r15 (- r0) (to_real (abs 92)))) (<= (- r10 (- (/ (- 25.247) r16)) r3) 74 (* (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (/ 25.247 (- r2)) r15 (- r0) (to_real (abs 92))))))
(assert (or (> (+ r14 (- 3110 r3)) (/ r28 (- 0.1618))) (= S1-0 S1-0 S1-0) (is_int 3110)))
(assert (or (<= (- r10 (- (/ (- 25.247) r16)) r3) 74 (* (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (/ 25.247 (- r2)) r15 (- r0) (to_real (abs 92)))) v27 v25))
(assert (or (> (+ r14 (- 3110 r3)) (/ r28 (- 0.1618))) (= S1-0 S1-0 S1-0) (<= (- r10 (- (/ (- 25.247) r16)) r3) 74 (* (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (/ 25.247 (- r2)) r15 (- r0) (to_real (abs 92))))))
(assert (or (distinct (- (+ 25.247 656 r5 r15)) (- r0) (/ (- 25.247) r16) 0.11 (- (- r2) (/ (- r0) r17) (- r11))) (<= (- r10 (- (/ (- 25.247) r16)) r3) 74 (* (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (/ 25.247 (- r2)) r15 (- r0) (to_real (abs 92)))) (is_int 3110)))
(assert (or (is_int 3110) (= S1-0 S1-0 S1-0) (distinct (- (+ 25.247 656 r5 r15)) (- r0) (/ (- 25.247) r16) 0.11 (- (- r2) (/ (- r0) r17) (- r11)))))
(assert (or (distinct (- (+ 25.247 656 r5 r15)) (- r0) (/ (- 25.247) r16) 0.11 (- (- r2) (/ (- r0) r17) (- r11))) (= S1-0 S1-0 S1-0) (is_int 3110)))
(assert (or (> (+ r14 (- 3110 r3)) (/ r28 (- 0.1618))) (is_int 3110) (= S1-0 S1-0 S1-0)))
(assert (or (= S1-0 S1-0 S1-0) (is_int 3110) (> (+ r14 (- 3110 r3)) (/ r28 (- 0.1618)))))
(assert (or (and (< 182 (- 58)) v3 (>= (mod 83 i7) 795) (= (/ r5 r4) (- r11)) (>= r19 (+ r14 (- 3110 r3)) 0.11 r4) (distinct 4695330 r3 (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (- r2) (* 176 (- r10 (- (/ (- 25.247) r16)) r3))) v48 (distinct (div (to_int 0.1618) 17) (abs 58)) (<= (to_real (- 592 (abs 61) 182)) (to_real (mod (to_int (+ 85 (- r10 85 (+ 25.247 656 r5 r15)) (+ (- r2) (- 25.247)))) (+ (div (+ 61 i6) (to_int 0.1618)) 708 (mod 61 i5) i12 (to_int 4695330)))) r10 r29)) (is_int 3110) (is_int r22)))
(assert (or (is_int 3110) (and (< 182 (- 58)) v3 (>= (mod 83 i7) 795) (= (/ r5 r4) (- r11)) (>= r19 (+ r14 (- 3110 r3)) 0.11 r4) (distinct 4695330 r3 (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (- r2) (* 176 (- r10 (- (/ (- 25.247) r16)) r3))) v48 (distinct (div (to_int 0.1618) 17) (abs 58)) (<= (to_real (- 592 (abs 61) 182)) (to_real (mod (to_int (+ 85 (- r10 85 (+ 25.247 656 r5 r15)) (+ (- r2) (- 25.247)))) (+ (div (+ 61 i6) (to_int 0.1618)) 708 (mod 61 i5) i12 (to_int 4695330)))) r10 r29)) v25))
(assert (or (> (+ r14 (- 3110 r3)) (/ r28 (- 0.1618))) (is_int 3110) (is_int r22)))
(assert (or v27 (and (< 182 (- 58)) v3 (>= (mod 83 i7) 795) (= (/ r5 r4) (- r11)) (>= r19 (+ r14 (- 3110 r3)) 0.11 r4) (distinct 4695330 r3 (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (- r2) (* 176 (- r10 (- (/ (- 25.247) r16)) r3))) v48 (distinct (div (to_int 0.1618) 17) (abs 58)) (<= (to_real (- 592 (abs 61) 182)) (to_real (mod (to_int (+ 85 (- r10 85 (+ 25.247 656 r5 r15)) (+ (- r2) (- 25.247)))) (+ (div (+ 61 i6) (to_int 0.1618)) 708 (mod 61 i5) i12 (to_int 4695330)))) r10 r29)) (and (< 182 (- 58)) v3 (>= (mod 83 i7) 795) (= (/ r5 r4) (- r11)) (>= r19 (+ r14 (- 3110 r3)) 0.11 r4) (distinct 4695330 r3 (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (- r2) (* 176 (- r10 (- (/ (- 25.247) r16)) r3))) v48 (distinct (div (to_int 0.1618) 17) (abs 58)) (<= (to_real (- 592 (abs 61) 182)) (to_real (mod (to_int (+ 85 (- r10 85 (+ 25.247 656 r5 r15)) (+ (- r2) (- 25.247)))) (+ (div (+ 61 i6) (to_int 0.1618)) 708 (mod 61 i5) i12 (to_int 4695330)))) r10 r29))))
(assert (or (<= (- r10 (- (/ (- 25.247) r16)) r3) 74 (* (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (/ 25.247 (- r2)) r15 (- r0) (to_real (abs 92)))) v27 (is_int 3110)))
(assert (or (and (< 182 (- 58)) v3 (>= (mod 83 i7) 795) (= (/ r5 r4) (- r11)) (>= r19 (+ r14 (- 3110 r3)) 0.11 r4) (distinct 4695330 r3 (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (- r2) (* 176 (- r10 (- (/ (- 25.247) r16)) r3))) v48 (distinct (div (to_int 0.1618) 17) (abs 58)) (<= (to_real (- 592 (abs 61) 182)) (to_real (mod (to_int (+ 85 (- r10 85 (+ 25.247 656 r5 r15)) (+ (- r2) (- 25.247)))) (+ (div (+ 61 i6) (to_int 0.1618)) 708 (mod 61 i5) i12 (to_int 4695330)))) r10 r29)) v25 (= S1-0 S1-0 S1-0)))
(assert (or (is_int 3110) (is_int r22) v25))
(assert (or v25 (and (< 182 (- 58)) v3 (>= (mod 83 i7) 795) (= (/ r5 r4) (- r11)) (>= r19 (+ r14 (- 3110 r3)) 0.11 r4) (distinct 4695330 r3 (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (- r2) (* 176 (- r10 (- (/ (- 25.247) r16)) r3))) v48 (distinct (div (to_int 0.1618) 17) (abs 58)) (<= (to_real (- 592 (abs 61) 182)) (to_real (mod (to_int (+ 85 (- r10 85 (+ 25.247 656 r5 r15)) (+ (- r2) (- 25.247)))) (+ (div (+ 61 i6) (to_int 0.1618)) 708 (mod 61 i5) i12 (to_int 4695330)))) r10 r29)) (and (< 182 (- 58)) v3 (>= (mod 83 i7) 795) (= (/ r5 r4) (- r11)) (>= r19 (+ r14 (- 3110 r3)) 0.11 r4) (distinct 4695330 r3 (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (- r2) (* 176 (- r10 (- (/ (- 25.247) r16)) r3))) v48 (distinct (div (to_int 0.1618) 17) (abs 58)) (<= (to_real (- 592 (abs 61) 182)) (to_real (mod (to_int (+ 85 (- r10 85 (+ 25.247 656 r5 r15)) (+ (- r2) (- 25.247)))) (+ (div (+ 61 i6) (to_int 0.1618)) 708 (mod 61 i5) i12 (to_int 4695330)))) r10 r29))))
(assert (or (is_int r22) v27 (distinct (- (+ 25.247 656 r5 r15)) (- r0) (/ (- 25.247) r16) 0.11 (- (- r2) (/ (- r0) r17) (- r11)))))
(assert (or (<= (- r10 (- (/ (- 25.247) r16)) r3) 74 (* (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (/ 25.247 (- r2)) r15 (- r0) (to_real (abs 92)))) (> (+ r14 (- 3110 r3)) (/ r28 (- 0.1618))) v25))
(assert (or (and (< 182 (- 58)) v3 (>= (mod 83 i7) 795) (= (/ r5 r4) (- r11)) (>= r19 (+ r14 (- 3110 r3)) 0.11 r4) (distinct 4695330 r3 (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (- r2) (* 176 (- r10 (- (/ (- 25.247) r16)) r3))) v48 (distinct (div (to_int 0.1618) 17) (abs 58)) (<= (to_real (- 592 (abs 61) 182)) (to_real (mod (to_int (+ 85 (- r10 85 (+ 25.247 656 r5 r15)) (+ (- r2) (- 25.247)))) (+ (div (+ 61 i6) (to_int 0.1618)) 708 (mod 61 i5) i12 (to_int 4695330)))) r10 r29)) (<= (- r10 (- (/ (- 25.247) r16)) r3) 74 (* (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (/ 25.247 (- r2)) r15 (- r0) (to_real (abs 92)))) (and (< 182 (- 58)) v3 (>= (mod 83 i7) 795) (= (/ r5 r4) (- r11)) (>= r19 (+ r14 (- 3110 r3)) 0.11 r4) (distinct 4695330 r3 (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (- r2) (* 176 (- r10 (- (/ (- 25.247) r16)) r3))) v48 (distinct (div (to_int 0.1618) 17) (abs 58)) (<= (to_real (- 592 (abs 61) 182)) (to_real (mod (to_int (+ 85 (- r10 85 (+ 25.247 656 r5 r15)) (+ (- r2) (- 25.247)))) (+ (div (+ 61 i6) (to_int 0.1618)) 708 (mod 61 i5) i12 (to_int 4695330)))) r10 r29))))
(assert (or v25 (is_int 3110) (is_int 3110)))
(assert (or (is_int r22) (and (< 182 (- 58)) v3 (>= (mod 83 i7) 795) (= (/ r5 r4) (- r11)) (>= r19 (+ r14 (- 3110 r3)) 0.11 r4) (distinct 4695330 r3 (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (- r2) (* 176 (- r10 (- (/ (- 25.247) r16)) r3))) v48 (distinct (div (to_int 0.1618) 17) (abs 58)) (<= (to_real (- 592 (abs 61) 182)) (to_real (mod (to_int (+ 85 (- r10 85 (+ 25.247 656 r5 r15)) (+ (- r2) (- 25.247)))) (+ (div (+ 61 i6) (to_int 0.1618)) 708 (mod 61 i5) i12 (to_int 4695330)))) r10 r29)) (> (+ r14 (- 3110 r3)) (/ r28 (- 0.1618)))))
(assert (or (and (< 182 (- 58)) v3 (>= (mod 83 i7) 795) (= (/ r5 r4) (- r11)) (>= r19 (+ r14 (- 3110 r3)) 0.11 r4) (distinct 4695330 r3 (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (- r2) (* 176 (- r10 (- (/ (- 25.247) r16)) r3))) v48 (distinct (div (to_int 0.1618) 17) (abs 58)) (<= (to_real (- 592 (abs 61) 182)) (to_real (mod (to_int (+ 85 (- r10 85 (+ 25.247 656 r5 r15)) (+ (- r2) (- 25.247)))) (+ (div (+ 61 i6) (to_int 0.1618)) 708 (mod 61 i5) i12 (to_int 4695330)))) r10 r29)) v25 (<= (- r10 (- (/ (- 25.247) r16)) r3) 74 (* (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (/ 25.247 (- r2)) r15 (- r0) (to_real (abs 92))))))
(assert (or (distinct (- (+ 25.247 656 r5 r15)) (- r0) (/ (- 25.247) r16) 0.11 (- (- r2) (/ (- r0) r17) (- r11))) (is_int r22) (is_int 3110)))
(assert (or v25 (<= (- r10 (- (/ (- 25.247) r16)) r3) 74 (* (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (/ 25.247 (- r2)) r15 (- r0) (to_real (abs 92)))) (> (+ r14 (- 3110 r3)) (/ r28 (- 0.1618)))))
(assert (or (> (+ r14 (- 3110 r3)) (/ r28 (- 0.1618))) v25 v27))
(assert (or (is_int r22) (and (< 182 (- 58)) v3 (>= (mod 83 i7) 795) (= (/ r5 r4) (- r11)) (>= r19 (+ r14 (- 3110 r3)) 0.11 r4) (distinct 4695330 r3 (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (- r2) (* 176 (- r10 (- (/ (- 25.247) r16)) r3))) v48 (distinct (div (to_int 0.1618) 17) (abs 58)) (<= (to_real (- 592 (abs 61) 182)) (to_real (mod (to_int (+ 85 (- r10 85 (+ 25.247 656 r5 r15)) (+ (- r2) (- 25.247)))) (+ (div (+ 61 i6) (to_int 0.1618)) 708 (mod 61 i5) i12 (to_int 4695330)))) r10 r29)) (is_int r22)))
(assert (or (distinct (- (+ 25.247 656 r5 r15)) (- r0) (/ (- 25.247) r16) 0.11 (- (- r2) (/ (- r0) r17) (- r11))) (and (< 182 (- 58)) v3 (>= (mod 83 i7) 795) (= (/ r5 r4) (- r11)) (>= r19 (+ r14 (- 3110 r3)) 0.11 r4) (distinct 4695330 r3 (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (- r2) (* 176 (- r10 (- (/ (- 25.247) r16)) r3))) v48 (distinct (div (to_int 0.1618) 17) (abs 58)) (<= (to_real (- 592 (abs 61) 182)) (to_real (mod (to_int (+ 85 (- r10 85 (+ 25.247 656 r5 r15)) (+ (- r2) (- 25.247)))) (+ (div (+ 61 i6) (to_int 0.1618)) 708 (mod 61 i5) i12 (to_int 4695330)))) r10 r29)) (= S1-0 S1-0 S1-0)))
(assert (or v25 v25 (is_int r22)))
(assert (or (distinct (- (+ 25.247 656 r5 r15)) (- r0) (/ (- 25.247) r16) 0.11 (- (- r2) (/ (- r0) r17) (- r11))) v25 (is_int 3110)))
(assert (or (= S1-0 S1-0 S1-0) (and (< 182 (- 58)) v3 (>= (mod 83 i7) 795) (= (/ r5 r4) (- r11)) (>= r19 (+ r14 (- 3110 r3)) 0.11 r4) (distinct 4695330 r3 (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (- r2) (* 176 (- r10 (- (/ (- 25.247) r16)) r3))) v48 (distinct (div (to_int 0.1618) 17) (abs 58)) (<= (to_real (- 592 (abs 61) 182)) (to_real (mod (to_int (+ 85 (- r10 85 (+ 25.247 656 r5 r15)) (+ (- r2) (- 25.247)))) (+ (div (+ 61 i6) (to_int 0.1618)) 708 (mod 61 i5) i12 (to_int 4695330)))) r10 r29)) (<= (- r10 (- (/ (- 25.247) r16)) r3) 74 (* (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (/ 25.247 (- r2)) r15 (- r0) (to_real (abs 92))))))
(assert (or (= S1-0 S1-0 S1-0) (> (+ r14 (- 3110 r3)) (/ r28 (- 0.1618))) (= S1-0 S1-0 S1-0)))
(assert (or (and (< 182 (- 58)) v3 (>= (mod 83 i7) 795) (= (/ r5 r4) (- r11)) (>= r19 (+ r14 (- 3110 r3)) 0.11 r4) (distinct 4695330 r3 (- (- r0) r16 (- (+ 25.247 656 r5 r15)) r0) (- r2) (* 176 (- r10 (- (/ (- 25.247) r16)) r3))) v48 (distinct (div (to_int 0.1618) 17) (abs 58)) (<= (to_real (- 592 (abs 61) 182)) (to_real (mod (to_int (+ 85 (- r10 85 (+ 25.247 656 r5 r15)) (+ (- r2) (- 25.247)))) (+ (div (+ 61 i6) (to_int 0.1618)) 708 (mod 61 i5) i12 (to_int 4695330)))) r10 r29)) v25 v27))
(check-sat)
(exit)
```

**Comment by ajreynol** [comment]
Thanks for the benchmark.  I assume you are running with --check-model?

CVC4 is throwing a spurious error for this benchmark. So in other words, this is a problem with the --check-model option + non-linear, but the answer "sat" and the model is correct. 

In more detail, it is complaining that an internally generated literal is not satisfied by the model but that literal's truth is justified internally in a non-standard way that is not compatible with --check-model. 

I will submit a fix for this soon.

**Comment by ajreynol** [comment]
This was fixed in #3525.
